### DISCUSSION TITLE
Insane COP values / SCOP calculation

### DISCUSSION THREAD

**Discussion by TheRealGreatOldOne** [raomin/ESPAltherma#431] [Q&A]
Hi,

Firstly, I’ve installed espaltherma on my heat pump - an EDLA08E2V3 - and I get data and I’m able to display this in homeassistant, and generate a COP value.

however, looking at the history of the generated COP value, it gets wild spikes where it appears to be breaking the laws of physics. 

![IMG_1959](https://github.com/raomin/ESPAltherma/assets/43447047/8144fc0e-08e3-4971-beba-df0aacdd6756)

I very much doubt that I really had a COP of 26 just before lunchtime today. I’m using the published formula from the GitHub article 

```
- name: "COP"      
  unique_id: "espaltherma_cop"      
  unit_of_measurement: 'COP'      
  state: "{% if is_state_attr('sensor.althermasensors','Operation Mode', 'Heating') and is_state_attr('sensor.althermasensors','Freeze Protection', 'OFF')  %}         
     {{           ((state_attr('sensor.althermasensors','Flow sensor (l/min)')| float * 0.06 * 1.16 * (state_attr('sensor.althermasensors','Leaving water temp. before BUH (R1T)') | float - state_attr('sensor.althermasensors','Inlet water temp.(R4T)')|float) )            /          (state_attr('sensor.althermasensors','INV primary current (A)')| float * states('sensor.solarman_meter_voltage') | float / 1000))          |round(2)        
     }}        
  {% else %} 0 {%endif%}"        

``` 

All of the data is coming from espaltherma bar the line voltage - which isn’t present in the config file for my model heat pump. I get this from my solar inverter instead.

any ideas why this is occurring? 

Secondly - I’ve search here and on the git pages for info on how to calculate the SCOP. The only answers I’ve seen are ‘it’s discussed here’ with a link… but the link doesn’t really explain. But as it’s a seasonal COP, or essentially the COP as measured over a set period of time - can we log individual COP values for a set period of time, and then average for the number of values generated during that period? This obviously will only work if there’s no ridiculously off bat values being generated. :)

any help would be appreciated.

**Comment by TheRealGreatOldOne** [comment]
Bumping to see if anyone has any idea on this. 

Jason

**Comment by vincent-bosche** [comment]
I was having the same issue.
I'm having a SmartMeter for my whole house. So I can guess what the consumption of the heatpump is.

So I reversed the calculation of the Power (W) and found that `3 * INV primary current (A) * 230V` will create a good Value.

With this value I get also a realistic COP of 3,5 - 4,5.
